DEV/org.openl.rules/src/org/openl/rules/context/RulesRuntimeContextFactory.java

```diff
@@ -1,7 +1,22 @@
 package org.openl.rules.context;
 
+import java.util.Locale;
+
 public class RulesRuntimeContextFactory {
+
+    private static final ThreadLocal<Locale> LOCALE_HOLDER = new ThreadLocal<>();
+
+    public static void setLocale(Locale locale) {
+        LOCALE_HOLDER.set(locale);
+    }
+
+    public static void clearLocale() {
+        LOCALE_HOLDER.remove();
+    }
+
     public static IRulesRuntimeContext buildRulesRuntimeContext() {
-        return new DefaultRulesRuntimeContext();
+        DefaultRulesRuntimeContext rulesRuntimeContext = new DefaultRulesRuntimeContext();
+        rulesRuntimeContext.setLocale(LOCALE_HOLDER.get());
+        return rulesRuntimeContext;

```

Comment:
How will it work if BA declared a Locale as a field in Datatype like the following:

```
Locale locale : context.locale
```

I assume, that the value from Datatype will override the default value in `IRulesRuntimeContext`. So, this improvement will not allow removing the custom interceptor with `Accept-Language` handling without fixing OpenL rules

Reply:
If you inject `Locale` directly to `IRulesRuntimeContext`, I suggest supporting the same approach if context variable was declared in Datatype (As I provided an example in the prev comment). Automate `Accept-Language` handling must be the same for all approaches for declaring context fields.

Reply:
This solution defines only a default value of the 'locale' property based on `Accept-Language` header. If somewhere it is overridden, so it becomes not default value.

Reply:
The question is, if the datatype 'locale' field was not sent, will null value be set in the context property? 
There should not be conflict between two approaches of defining 'locale' via HTTP `Accept-Language` header and sending of 'locale' field in the request.

Reply:
Added handling of null (undefined) values for 'locale'. See tests.

Reply:
For instance, you may implement default service call before the interceptor and implement inject of Accept-Language value there. So, if someone will need it, they will annotate method with this interceptor in annotation template interface